File: python/pilas.py
class Pila:
    def __init__(self):
        self.libros = []

    def esta_vacia(self):
        return len(self.libros) == 0

    def apilar(self, libro):
        self.libros.append(libro)

    def desapilar(self):
        if self.esta_vacia():
            return None
        return self.libros.pop()


def apilar_libros(pila, libros):
    for libro in libros:
        pila.apilar(libro)


def desapilar_libros(pila):
    while not pila.esta_vacia():
        libro = pila.desapilar()
        print(f"Desapilando: {libro}")


pila_libros = Pila()
libros = ['Libro 1', 'Libro 2', 'Libro 3']

apilar_libros(pila_libros, libros)
desapilar_libros(pila_libros)

File: dart/pilas.dart
class Pila {
  List<String> _libros = [];

  bool estaVacia() {
    return _libros.isEmpty;
  }

  void apilar(String libro) {
    _libros.add(libro);
  }

  String? desapilar() {
    if (estaVacia()) {
      return null;
    }
    return _libros.removeLast();
  }
}

void apilarLibros(Pila pila, List<String> libros) {
  for (String libro in libros) {
    pila.apilar(libro);
  }
}

void desapilarLibros(Pila pila) {
  while (!pila.estaVacia()) {
    String? libro = pila.desapilar();
    print('Desapilando libro: $libro');
  }
}

void main() {
  Pila pilaLibros = Pila();
  List<String> libros = ['Libro 1', 'Libro 2', 'Libro 3'];

  apilarLibros(pilaLibros, libros);

  desapilarLibros(pilaLibros);
}
